src/frontends/tensorflow/src/op/reverse.cpp

```diff
@@ -3,47 +3,129 @@
+    }
+
+    auto input_shape = make_shared<ShapeOf>(batched_input, element::i32);
+    auto seq_lenghts = compute_sequence_lengths(input_shape, batch_axis, seq_axis);
+    auto reverse_sequence = make_shared<ReverseSequence>(batched_input, seq_lenghts, batch_axis, seq_axis)->output(0);
+
+    if (unsqueeze_axes.size() > 0) {
+        // remove earlier added additional dimensions from the result
+        auto squeeze_axes_const = make_shared<Constant>(element::i64, Shape{unsqueeze_axes.size()}, unsqueeze_axes);
+        reverse_sequence = make_shared<Squeeze>(reverse_sequence, squeeze_axes_const);
+    }
+
+    set_node_name(node.get_name(), reverse_sequence.get_node_shared_ptr());
+    return {reverse_sequence};
+}
 
 OutputVector translate_reverse_op(const NodeContext& node) {
+    // The second input of Reverse is a boolean vector.
+    // True elements correspond the axes along which
+    // elements of the input tensor are reversed
+    default_op_checks(node, 2, {"Reverse"});
     auto input = node.get_input(0);
-    auto axes = node.get_input(1);
-
-    auto axes_const = dynamic_pointer_cast<Constant>(axes.get_node_shared_ptr());
-    TENSORFLOW_OP_VALIDATION(node, axes_const != nullptr, "Axes input must be constant.");
-    TENSORFLOW_OP_VALIDATION(node, axes_const->get_shape().size() == 1, "Axes input must be 1D.");
-    TENSORFLOW_OP_VALIDATION(node, axes_const->get_shape()[0] == 1, "Axes input must have only one value.");
-    auto seq_axis = axes_const->cast_vector<int64_t>().at(0);
-    int64_t batch_axis = !seq_axis;
-
-    Output<Node> seq_lengths;
-    if (input.get_partial_shape().is_static()) {
-        auto in_shape = input.get_shape();
-        seq_lengths = make_shared<Constant>(element::i64, Shape{in_shape[batch_axis]}, in_shape[seq_axis]);
-    } else {
-        auto shape = make_shared<ShapeOf>(input);
-        auto one = make_shared<Constant>(element::i64, Shape{1}, 1);
-        auto gather_batch = make_shared<Gather>(shape,
-                                                make_shared<Constant>(element::i64, Shape{1}, batch_axis),
-                                                make_shared<Constant>(element::i64, Shape{1}, 0));
-        auto gather_seq = make_shared<Gather>(shape,
-                                              make_shared<Constant>(element::i64, Shape{1}, seq_axis),
-                                              make_shared<Constant>(element::i64, Shape{1}, 0));
-        auto broadcast = make_shared<Broadcast>(one, gather_batch);
-        seq_lengths = make_shared<Multiply>(broadcast, gather_seq);
+
+    std::vector<bool> dims;
+    get_const_input(node, 1, &dims);
+
+    // collect axes along which to reverse
+    std::vector<int64_t> axes;
+    for (int64_t ind = 0; ind < static_cast<int64_t>(dims.size()); ++ind) {
+        if (dims[ind]) {
+            axes.push_back(ind);
+        }
     }
+    TENSORFLOW_OP_VALIDATION(node,
```

Comment:
nit: I don't think you need this validation here. You already have it inside `translate_reverse_base_op`, plus in case `axis == 0` it will skip the reverse instead of throwing.

Reply:
Great catch! Fixed.